### ISSUE TITLE
python with mixed types in table crashes

### ISSUE THREAD

**Issue by scottdraves** [twosigma/beakerx#6656]
```
TableDisplay([{'y1':4, 'm3':2, 'z2':1}, {'m3':4, 'z2':'z'}])
```
followup to https://github.com/twosigma/beakerx/issues/6638

**Comment by scottdraves** [comment]
this works in groovy:
```
List list = new ArrayList();
list.add(["x" : 1, "y" : 2]);
list.add(["x" : 3, "y" : 'a']);
list
```

**Comment by jmsdnns** [comment]
This _feels_ similar to an issue we solved on the java side that had to do with how column types are determined.

This is the relevant commit: https://github.com/twosigma/beakerx/commit/d2310862d3e4e6bbd77c186f593db1b0189e8362

Based on the above snippet, I could imagine the code reading numbers, assigning a number type to the column, and then the type fails when it encounters a string. If this speculation is correct, a similar fix to what we did in the commit above could probably work here.

The gist of the solution is to scan every value for any columns that do not have an assigned type of "string". If we find mismatched types or we find strings, we set the column type to string, which will safely display everything.
